imap_processing/cdf/utils.py

```diff
@@ -66,7 +67,11 @@ def write_cdf(dataset: xr.Dataset):
     start_time = np.datetime_as_string(dataset["epoch"].values[0], unit="D").replace(
         "-", ""
     )
-    version = f"v{int(dataset.attrs['Data_version']):03d}"  # vXXX
+    r = re.compile(r"v\d{3}")
+    if r.match(dataset.attrs["Data_version"]) is None:
```

Comment:
`r.match` will raise a TypeError if `dataset.attrs["Data_version"]` is not a string. Is this a possibility?

Reply:
It is a possibility but I think it would be an incorrect usage of dataset attributes. But, I can add a catch here if it's an int.